ofp4/p4include/of_model.p4

```diff
@@ -152,99 +157,201 @@ header Vlan {
     bit<12> vid;
 }
 
+@of_prereq("mpls")
 header Mpls {
+    @name("mpls_label")
     bit<32> label;              // Label (low 20 bits).
+
+    @name("mpls_tc")
     bit<8> tc;                  // Traffic class (low 3 bits).
+
+    @name("mpls_bos")
     bit<8> bos;                 // Bottom of Stack (low bit only).
+
+    @name("mpls_ttl")
     bit<8> ttl;                 // Time to live.
 }
 
 const bit<8> FRAG_ANY = 1 << 0;   // Set for any IP fragment.
 const bit<8> FRAG_LATER = 1 << 1; // Set for IP fragment with nonzero offset.
 
+@of_prereq("ipv4")
```

Comment:
Do you want to add a comment describing the annotation?

Reply:
Good idea, done.